### PR TITLE
Eligibility Fixes

### LINKED ISSUE #7308: Select list default selection of generated item
### Describe the bug

[This PR](https://github.com/openemr/openemr/pull/6659), specifically [this line](https://github.com/openemr/openemr/blob/328c15608c3c84fcd9e7d1ac7a3973ff9b311466/library/options.inc.php#L183) changes prior behavior by selecting the first item in a list by default. This creates havoc with multiple-select lists where a generated "Unassigned" option should not be selected by default.

### To Reproduce
1. Edit Demographics->Choices
2. Select a different Patient Categories option, e.g. "Group III"
3. Save
4. Edit Demographics->Choices
5. Check which Patient Categories option(s) are selected.

### LINKED ISSUE #7773: Repair OA Eligibility
### Describe the bug
Fetch request doesn't run.
Credentials size limit.
MIME response parsing no longer works
- Browser:
- OpenEMR version: since v7.0.0
- Operating system:

### LINKED ISSUE #7776: Encounter Review shows unknown in tab title.
### Describe the bug
Title says it all. There isn't a title tag in forms.php.

### PR DESCRIPTION
**Pull request by sjpadgett**
This is actually a pretty nice and still useful feature. 
Thanks for the help @stephenwaite 

Fixes #7773 
Fixes #7308 
Fixes #7776 

#### Changes proposed in this pull request:
Change eligibility password and login to use SFTP inputs. 
Rework MIME parsing.
Fix conditional preventing any request calls.
Fix default unassigned for multi select if not empty.

### CODE REVIEW

**Comment by bradymiller** [comment]
2DV-2

File: src/Billing/EDI270.php [hunk 1]
```diff
@@ -856,9 +859,9 @@ public static function requestEligibility($partner = '', $x12_270 = '')
         $formBody = $response->body();
```
Comment (stephenwaite) [inline src/Billing/EDI270.php:859]: hi @sjpadgett , I'm getting `Error:Http Error: Bad Request : 400` on testing. How do I see the validation errors that are in `$formBody`?

Comment (sjpadgett) [reply src/Billing/EDI270.php:859]: I'm getting same thing today. Need to check with user if he to if not working. 
Uncheck
image

image

File: library/options.inc.php [hunk 2]
```diff
@@ -177,14 +177,22 @@ function generate_select_list(
...
-            // should be provide a truley empty option.
+            // should be provided a truly empty option.
             $_options[] = [];
         } else {
-            $_options[] = [
-                'label' => $selectEmptyName,
-                'value' => '',
-                'isSelected' => true,
-            ];
+            if ($multiple && !empty($currvalue)) {
```
Comment (adunsulag) [inline library/options.inc.php:183]: This looks good @sjpadgett. Haven't pulled it down to test, but it looks like it will fix the problem.

**Comment by sjpadgett** [comment]
I'm merging this then backport unless anyone has issues.

**Comment by stephenwaite** [comment]
would like to get a successful test of the eligibility

**Comment by sjpadgett** [comment]
I already checked with Moussa and he tested again yesterday fine.
You have to remember that provider/NPI and perhaps other params need to match OA account.
Do you have you own account?

**Comment by stephenwaite** [comment]
pasting the 270 string into their curl example works but the oeHttp post is giving a 400

**Comment by stephenwaite** [comment]
how would you capture the 999 that's returned with any errors?

**Comment by sjpadgett** [comment]
image

send me the curl example. I receive the response from OA see above.

**Comment by stephenwaite** [comment]
that option is broken?

**Comment by sjpadgett** [comment]
AAA*Y**42*R~HL*2*1*21*1~NM1*1P*2*LOCATIONB*****XX*1234567890~HL*3*2*22*0~TRN*1*502227609-20241023*9OFFICALLY~NM1*IL*1*EL-HALLAK*HADI****MI*910001626470~REF*EJ*5~DMG*D8*20100618*M~DTP*291*D8*20241023~EB*V~MSG*INVALID RESPONSE RECEIVED FROM THE PAYER~SE*16*402227609~GE*1*402227609~IEA*1*402227609~
AAA Segments:
Array
(
[0] => Array
(
[request_ind] => Y
[reason_code] => 42 : Unable to respond at current time.
[follow_up] => R : Resubmission allowed
)

)
Subscriber Member: HADI EL-HALLAK Member Id: 910001626470 ---*

Benefit Type: Cannot Process
Message: INVALID RESPONSE RECEIVED FROM THE PAYER

option unchecked

**Review by stephenwaite — APPROVED** [review]
smaller_excited_elephant

**Comment by sjpadgett** [comment]
good work @stephenwaite merging

**Pull request merged**